lib/resty/rediscluster.lua

```diff
@@ -137,6 +189,157 @@ local function split(s, delimiter)
+    -- 72d111446f11ee900df33f8a91425bb6533be0b6 172.26.0.23:6379@16379 master - 0 1718355187000 3 connected 10923-16383
+    -- 438d9bc4c865e342a0a6dd1b560ac5d3f0c92662 172.26.0.22:6379@16379 master - 0 1718355186902 2 connected 5461-10922
+    -- 0e0da2991a4f713691c209ba2b7467d2a8223e9c 172.26.0.24:6379@16379 slave 72d111446f11ee900df33f8a91425bb6533be0b6 0 1718355187920 3 connected
+    -- 86bc3b462782f48ee2b56939107744a792dc3575 172.26.0.21:6379@16379 myself,master - 0 1718355182000 1 connected 0-5460
+    local node_lst = split(nodes, char(10))  -- char(10) is "\n"
+    for _, node in ipairs(node_lst) do
+        local node_info = split(node, char(32))  -- char(32) is " "
+        if #node_info > 2 then
+            local is_master = string_find(node_info[3], "master", 1, true) ~= nil
+            if is_master then
+                -- 172.26.0.21:6379
+                local ip_port_str = split(node_info[2], char(64))[1]  -- char(64) is "@"
+                local ip_port = split(ip_port_str, char(58))  -- char(58) is ":"
+                table_insert(master_list, {
+                    ip = ip_port[1],
+                    port = tonumber(ip_port[2])
+                })
+            end
+        end
+    end
+
+    master_nodes[self.config.name .. "master_list"] = master_list
+
+    return true
+end
+
+
+local function parse_slots(self, redis_client)
+    if not redis_client then
+        return nil, "redis_client can not be nil"
+    end
+
+    local slots_data, err = redis_client:cluster("slots")
+    if not slots_data then
+        return nil, "failed to fetch slots: " .. err
+    end
+
+    -- mapping from slots to IPs and/or hostnames of servers
+    local slots = new_tab(0, 16384)
+
+    -- cahce only master nodes
+    local master_list = {}
+
+    -- cache the complete list of servers present in cluster
+    -- while the cluster is resharding (adding/removing nodes)
+    -- this can differ from self.config.serv_list
+    local servers = { serv_list = {} }
+
+    for i = 1, #slots_data do
+        -- Redis 6: { 0, 5460, { "172.26.0.21", 6379, "b050a28d860201c487891ecde77efc8a3935ea51" }, { "172.26.0.25", 6379, "20d96bc4a640fd244699e29274b2c94962def80e" } }
+        -- Redis 7: { 0, 5460, { "rc-node-1", 6379, "3a4f2814802ca25a5c92dfc167674802126557d3", { "ip", "172.26.0.11" } }, { "rc-node-5", 6379, "f14b2cf710613f517374aefe7b84d9f013102ebe", { "ip", "172.26.0.15" } } }
+        -- Redis 7: { 0, 5460, { "172.26.0.11", 6379, "64ff3bc4e8f4587c92bcfaa6c085e67417d44d5d", {} }, { "172.26.0.15", 6379, "9a63c473b928203b141d20a46be6784eb320ba0c", {} } }
+        local sub_data = slots_data[i]
+
+        -- item 1 and 2 are the start slot and end slot
+        local start_slot, end_slot = sub_data[1], sub_data[2]
+
+        -- cache the servers responsible for [start_slot, end_slot]
+        local sub_servers = { serv_list = {} }
```

Comment:
What does `sub_servers` stand for?  What does this variable name mean?

Reply:
The `sub_servers` refer to a list of servers responsible for slots `[start_slot, end_slot]`.

Will rename `sub_data` to `slot_range_data`, and `sub_servers` to `slot_range_servers`.

Reply:
I think I'm missing what does `sub` mean in this context? Like there's _master_ server and this is _slave_ ?